File: yunva17/Week_22/BOJ_2170_선긋기/BOJ_2170_선긋기.js
let input = require('fs')
  //  .readFileSync('/dev/stdin')
  .readFileSync('Week_22/BOJ_2170_선긋기/input.txt')
  .toString()
  .trim()
  .split('\n');

let lines = input.slice(1).map((v) => v.split(' ').map(Number));

lines = lines.sort((a, b) => (a[0] === b[0] ? a[1] - b[1] : a[0] - b[0]));

let answer = lines[0][1] - lines[0][0];
let [, maxEnd] = lines[0];

for (let i = 1; i < lines.length; i++) {
  let [start, end] = lines[i];
  if (start < maxEnd) {
    if (maxEnd < end) {
      answer += end - maxEnd;
    }
  } else {
    answer += end - start;
  }

  maxEnd = Math.max(maxEnd, end);
}

console.log(answer);

File: l-umi/Week_22/BOJ_2170_선_긋기.py
import sys
input = sys.stdin.readline

N = int(input())
line = []
for n in range(N):
    x, y = map(int, input().split())
    line.append((x,y))
line.sort()
# print(line)
####################

start = line[0][0]
end = line[0][1]
answer = 0
for i in range(1,N):
    if end < line[i][0]:
        answer += end-start
        start, end = line[i]
    else:
        end = max(end, line[i][1])
    # print(start, end, answer)

answer += end-start
print(answer)
